tests/controllers/feature/feature_test.go

```diff
@@ -0,0 +1,187 @@
+		Group:   "management.cattle.io",
+		Version: "v3",
+		Kind:    "NodeDriver",
+	})
+	assert.NoError(s.T(), err)
+
+	// The feature controller checks the harvester NodeDriver, so it needs to exist
+	n, err := s.wranglerContext.Mgmt.NodeDriver().Create(&v3.NodeDriver{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "harvester",
+		},
+	})
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), n)
+
+	// Initialize Features
+	features.InitializeFeatures(s.wranglerContext.Mgmt.Feature(), "")
+}
+
+func (s *FeatureTestSuite) TestHarvesterFeature() {
+	// If the harvester feature is enabled, it needs to enable the NodeDriver as well
+	n, err := s.wranglerContext.Mgmt.NodeDriver().Get("harvester", v1.GetOptions{})
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), n)
+
+	nCopy := n.DeepCopy()
+	nCopy.Spec.Active = false
+	_, err = s.wranglerContext.Mgmt.NodeDriver().Update(nCopy)
+	assert.NoError(s.T(), err)
+
+	s.triggerSync("harvester")
+
+	assert.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+		nd, err := s.wranglerContext.Mgmt.NodeDriver().Get("harvester", v1.GetOptions{})
+		assert.NoError(c, err)
+		assert.True(c, nd.Spec.Active)
+	}, duration, tick, "harvester feature has not been enabled")
+
+}
+
+func (s *FeatureTestSuite) TestHarvesterBaremetalFeature() {
+	// Set harvester baremetal feature to true
+	features.GetFeatureByName("harvester-baremetal-container-workload").Set(true)
+
+	// Harvester feature should get set if baremetal feature is enabled
+	h, err := s.wranglerContext.Mgmt.Feature().Get("harvester", v1.GetOptions{})
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), h)
+
+	hCopy := h.DeepCopy()
+	h.Spec.Value = nil
+	_, err = s.wranglerContext.Mgmt.Feature().Update(hCopy)
+	assert.NoError(s.T(), err)
+
+	// Baremetal feature gets annotation added if it's not there
+	hf, err := s.wranglerContext.Mgmt.Feature().Get("harvester-baremetal-container-workload", v1.GetOptions{})
+	assert.NoError(s.T(), err)
+	assert.NotNil(s.T(), hf)
+
+	harvesterFeature := hf.DeepCopy()
+	harvesterFeature.Annotations = nil
+	_, err = s.wranglerContext.Mgmt.Feature().Update(harvesterFeature)
+	assert.NoError(s.T(), err)
+
+	// Check that feature.cattle.io/experimental gets set to true by the controller
+	assert.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+		f, err := s.wranglerContext.Mgmt.Feature().Get("harvester-baremetal-container-workload", v1.GetOptions{})
+		assert.NoError(c, err)
+		assert.Equal(c, v3.ExperimentalFeatureValue, f.Annotations[v3.ExperimentalFeatureKey])
+	}, duration, tick, "annotation feature.cattle.io/experimental has not changed to true")
+
+	// Check that harvester feature gets set
+	assert.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+		nd, err := s.wranglerContext.Mgmt.Feature().Get("harvester", v1.GetOptions{})
+		assert.NoError(c, err)
+		assert.NotNil(c, nd.Spec.Value)
+	}, duration, tick, "harvester feature has not been enabled")
```

Comment:
Could these two be compressed into a single assertion, not just that `ns.Spec.Value` is not `nil` but that it's true?